Paintroid/src/main/java/org/catrobat/paintroid/dialog/LayersDialog.java

```diff
@@ -0,0 +1,483 @@
+import android.text.InputType;
+import android.util.Log;
+import android.view.View;
+import android.view.WindowManager;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.AdapterView;
+import android.widget.AdapterView.OnItemClickListener;
+import android.widget.AdapterView.OnItemLongClickListener;
+import android.widget.EditText;
+import android.widget.GridView;
+import android.widget.ImageButton;
+import android.widget.SeekBar;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import org.catrobat.paintroid.MainActivity;
+import org.catrobat.paintroid.PaintroidApplication;
+import org.catrobat.paintroid.R;
+import org.catrobat.paintroid.command.implementation.LayerCommand;
+import org.catrobat.paintroid.eventlistener.ChangeActiveLayerEventListener;
+import org.catrobat.paintroid.eventlistener.RefreshLayerDialogEventListener;
+import org.catrobat.paintroid.tools.Layer;
+import org.catrobat.paintroid.ui.button.LayersAdapter;
+
+import java.util.ArrayList;
+
+public final class LayersDialog extends BaseDialog implements OnItemClickListener,
+	OnItemLongClickListener, DialogInterface.OnDismissListener,
+		SeekBar.OnSeekBarChangeListener, RefreshLayerDialogEventListener, ChangeActiveLayerEventListener {
+
+	private static final String NOT_INITIALIZED_ERROR_MESSAGE = "LayerDialog has not been initialized. Call init() first!";
+
+	private static LayersDialog instance;
+
+	private LayersAdapter mLayerButtonAdapter;
+    private ImageButton mNewLayerButton;
+    private ImageButton mDeleteLayerButton;
+	private ImageButton mNameLayerButton;
+	private ImageButton mVisibleLayerButton;
+	private ImageButton mLockLayerButton;
+    private ImageButton mMergeLayerButton;
+	private MainActivity mParent;
+    private Layer mCurrentLayer;
+	private Context mContext;
+    private Layer firstLayertoMerge;
+    private Boolean mergeClicked;
+	private SeekBar mOpacitySeekbar;
+	private TextView mOpacitySeekbarLabel;
+
+	public LayersAdapter getAdapter() {
+		return mLayerButtonAdapter;
+	}
+
+	private LayersDialog(Context context, Bitmap first_layer) {
+		super(context);
+		mContext = context;
+		mParent = (MainActivity) context;
+        mLayerButtonAdapter = new LayersAdapter(context,
+				PaintroidApplication.openedFromCatroid, first_layer);
+	}
+
+	public static LayersDialog getInstance() {
+		if (instance == null) {
+			throw new IllegalStateException(NOT_INITIALIZED_ERROR_MESSAGE);
+		}
+		return instance;
+	}
+
+	void InitCurrentLayer() {
+		if(mLayerButtonAdapter == null) {
+			mLayerButtonAdapter = new LayersAdapter(mContext,
+					PaintroidApplication.openedFromCatroid, PaintroidApplication.drawingSurface.getBitmapCopy());
+		}
+		mCurrentLayer = mLayerButtonAdapter.getLayer(0);
+		if (mCurrentLayer != null) {
+			selectLayer(mCurrentLayer);
+			return;
+		}
+		Log.d("DEBUG", "CURRENT LAYER NOT INITIALIZED");
+
+	}
+
+	public static void init(MainActivity mainActivity, Bitmap first_layer) {
```

Comment:
I would add a check if instance is != null, because with this, everyone can overwrite the current instance if its not null. 
Or only create a new instance in the getInstance methode, if instance == null